test/test_ledger.py

```diff
@@ -34,41 +34,41 @@ def test_pin(self):
             result = self.do_command(self.dev_args + ['promptpin'])
             self.assertIn('error', result)
             self.assertIn('code', result)
-            self.assertEqual(result['error'], 'The Ledger Nano S does not need a PIN sent from the host')
+            self.assertEqual(result['error'], 'The Ledger Nano S and X do not need a PIN sent from the host')
```

Comment:
Can you try to never modify the errors? It's a CLI software so software that depends on it relies on parsing error strings.

Reply:
These particularly are not problematic for us, but it may be for others.

Reply:
I mean it's a terrible precedent. Please keep this in mind in further PRs: https://lkml.org/lkml/2012/12/23/75

> If a change results in user programs breaking, it's a bug in the
kernel. We never EVER blame the user programs. How hard can this be to
understand?

Reply:
Hmmm. Errors would ideally return error numbers as well and warn devs to not depend on error message strings.

Reply:
> Hmmm. Errors would ideally return error numbers as well and warn devs to not depend on error message strings.

Yes, indeed that's the clean way to do this: https://github.com/bitcoin-core/HWI/issues/180

It however only partly addresses the more general issue when not error messages, are broken. That being said I may be overly cautious here and maybe the maintainers are experienced enough with maintaining other software that builds on top of other software and libraries to understand that breaking even the smallest things are extremely dangerous and can result in far reaching consequences. This is even more true when we talk about a software that handles money.

Reply:
> If a change results in user programs breaking

Ugh no, people really shouldn't be relying on exact error messages. If error messages become stale that means they are invalid for manual consumption and users will be unable to fix their issue. Error codes please.

Reply:
Are you all blind or have you not tested HWI yourself? Literally every error has an associated error code already. The existence of the an error code is tested in the line directly above this one, and the code itself is tested in the line after this.

Reply:
I see you have error codes on normal commands, however you don't on `enumerate`. I'm only parsing enumerate errors for example, that's why I didn't see them before.

```
[{"type": "coldcard", "path": "...", "error": "Could not open client or get fingerprint information: 'NoneType' object is not callable"}]
```